src/main/java/io/kestra/plugin/fs/vfs/Upload.java

```diff
@@ -32,12 +34,14 @@ public Upload.Output run(RunContext runContext) throws Exception {
         try (StandardFileSystemManager fsm = new StandardFileSystemManager()) {
             fsm.init();
 
+            var renderedFrom = runContext.render(this.from);
+            var renderedTo = this.to != null ? runContext.render(this.to) : renderedFrom.substring(renderedFrom.lastIndexOf('/'));
```

Comment:
check if lastIndexOf == -1

Reply:
In fact what's missing is to check that from is a Ketra URI, with that check we can blindly indexOd.